ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java

```diff
@@ -127,6 +156,56 @@ void processDepositTopsUpValidatorBalanceWhenPubkeyIsFoundInRegistry()
         beaconState.getBalances().get(originalValidatorBalancesSize - 1));
   }
 
+  @Test
+  void processDepositHandlesDepositWithInvalidPublicKey() throws BlockProcessingException {
+    // The following deposit uses a "rogue" public key that is not in the G1 group
+    BLSPublicKey pubkey =
+        BLSPublicKey.fromBytesCompressed(
+            Bytes.fromHexString(
+                "0x9378a6e3984e96d2cd50450c76ca14732f1300efa04aecdb805b22e6d6926a85ef409e8f3acf494a1481090bf32ce3bd"));
+    Bytes32 withdrawalCredentials =
+        Bytes32.fromHexString("0x79e43d39ee55749c55994a7ab2a3cb91460cec544fdbf27eb5717c43f970c1b6");
+    UnsignedLong amount = UnsignedLong.valueOf(1000000000L);
+    BLSSignature signature =
+        BLSSignature.fromBytes(
+            Bytes.fromHexString(
+                "0xddc1ca509e29c6452441069f26da6e073589b3bd1cace50e3427426af5bfdd566d077d4bdf618e249061b9770471e3d515779aa758b8ccb4b06226a8d5ebc99e19d4c3278e5006b837985bec4e0ce39df92c1f88d1afd0f98dbae360024a390d"));
+    DepositData depositInput =
+        new DepositData(new DepositMessage(pubkey, withdrawalCredentials, amount), signature);
+
+    // Add the deposit to a Merkle tree so that we can get the root to put into the state
+    MerkleTree depositMerkleTree = new OptimizedMerkleTree(Constants.DEPOSIT_CONTRACT_TREE_DEPTH);
+    depositMerkleTree.add(depositInput.hash_tree_root());
+
+    // Create the state and insert the Merkle root of the deposit data
+    BeaconState beaconState =
+        createBeaconState()
+            .updated(
+                state ->
+                    state.setEth1_data(
+                        new Eth1Data(
+                            depositMerkleTree.getRoot(), UnsignedLong.valueOf(1), Bytes32.ZERO)));
+
+    SSZMutableList<DepositWithIndex> deposits =
+        SSZList.createMutable(DepositWithIndex.class, Constants.MAX_DEPOSITS);
+    deposits.add(
+        new DepositWithIndex(depositMerkleTree.getProof(0), depositInput, UnsignedLong.valueOf(0)));
+
+    int originalValidatorRegistrySize = beaconState.getValidators().size();
+    int originalValidatorBalancesSize = beaconState.getBalances().size();
+
+    // Attempt to process deposit with above data. We expect to fail, but not throw an exception.
+    beaconState =
+        beaconState.updated(state -> BlockProcessorUtil.process_deposits(state, deposits));
+
+    assertTrue(
+        beaconState.getValidators().size() == originalValidatorRegistrySize,
+        "The validator was added to the validator registry.");
+    assertTrue(
+        beaconState.getBalances().size() == originalValidatorBalancesSize,
+        "The balance was added to the validator balances.");
```

Comment:
(optional) Maybe we should verify that all of the balances are unchanged.

Reply:
Good idea. Done.